sample_usage/draw_cfg.py

```diff
@@ -28,13 +28,13 @@ def display(cfgs: Dict[NodeNG, ControlFlowGraph],
         else:
             continue
         with graph.subgraph(name=f'cluster_{id(node)}') as c:
-            _visit(cfg, cfg.start, c, set())
+            _visit(cfg.end, c, set())
```

Comment:
Why did you switch this from start to end?

Reply:
This is because I wanted to capture every block in the control flow graph (even the unreachable ones). My understanding is that using this traversal method starting from `start` won’t capture unreachable blocks, while starting the traversal from `end` will ensure that.

Reply:
We can't assume there's a path from every block to the end node. For example,

```python
for i in range(10):
  print('reachable')
  break
  print('unreachable and doesn't reach end')
  break
  print('also unreachable, but reaches end')
```

Reply:
This is an interesting example, I ran this code on `draw_cfg.py` and this is what it returned:
<img width="789" alt="Screen Shot 2019-08-08 at 3 40 55 PM" src="https://user-images.githubusercontent.com/18351436/62732670-f0854200-b9f2-11e9-99f3-fffdecd632a5.png">
 
At first, I got totally confused but then it made sense and this is how the CFG actually should look like. 

The reason I think that there is a path from the `end` block to every block in the CFG is because of how unreachable blocks are stored. These blocks are added as predecessors to blocks that are reachable and they are not successors to the `start` block (this makes them unreachable). However, if we traverse the CFG from the `end` to the `start` we will be doing so by visiting every ancestor (i.e predecessor) block of `end`. This means that reachable blocks will be visited as well as unreachable blocks.

Reply:
Ah yes, I forgot that `break` and `return` do a link to the CFG end block. It does seem true that all blocks, at least for now, have a successor that is reachable.

However, this might change in the future (e.g., if we do some of the removal of edges based on constant propagation). I'd rather not change the algorithm more than once. The reason I said to use an attribute to store the unreachable nodes is so that we don't have to rely on these nodes being reachable from any other node in the graph.